.github/workflows/windows.yml

```diff
@@ -10,22 +10,17 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       matrix:
-        arch: [x64]
-        toolset: [v142]
-        msvs: ['Visual Studio 16 2019']
-        vs-versions: ['[16.0,16.30)']
-        os: [windows-2019]
+        arch: [Win32, x64]
+        toolset: [v143]
+        msvs: ['Visual Studio 17 2022']
+        vs-versions: ['[17.0,17.30)']
+        os: [windows-2022]
         include:
           - arch: Win32
-            toolset: v141
+            toolset: v142
```

Comment:
why 142 not 143 here?

Reply:
Don't we need to support both `v142` for `MS VS 2019` and `v143` for `MS VS 2022`?

Reply:
We could probably drop 2019 soon, but sure.

But I meant to ask about specifically what the result of this is, as I don't really understand the purpose of having some stuff in the `matrix:` and some stuff in `include:` within `matrix`... like why isn't it `toolset: [v142, v143]` above? Why are these specific values in the `include` bit?

Reply:
Because of Github syntax it will generate variants `v142` for `MS VS 2022` and `v143` for `MS VS 2019` as well if defined `toolset: [v142, v143]` and `msvs: ['Visual Studio 17 2022', 'Visual Studio 16 2019']`.